### ISSUE TITLE
[Proposal]Support dynamic buckets setting in dynamica parition

### ISSUE THREAD

**Issue by WingsGo** [apache/doris#3590]
**Is your feature request related to a problem? Please describe.**
When user create a dynamic partition table, they will sets a lot of parameters, and the table's daily datas may increase or decrease over time.
In order to  provide more convient usage to users, we can introudce a strategy to dynamic setting partitions buckets for users and keep the default behavior, if they don't want to set the buckets num, they can choose our strategy to dynamic setting buckets, I will investigate other OLAP engine to introduce dynamic setting buckets to Doris.

**Comment by marising** [comment]
This is very useful for most users, with a higher number of users unable to set the correct number of buckets.

**Comment by WingsGo** [comment]
Hi, @morningman @kangkaisen  The problem's nature is a math model to predicate the trends in data change, but a good model is get by data, we cannot give a unify model to predicate all situation,.So I decide to provide a simple model with simple sequential average method.And expose the interface, If we have another strategy, we can implement it and choose different strategy to adjust different situation, and I will give a param named `dynamic_partition.buckets_strategy` , if it is not specified, I will use the default simple mode to predicate the buckets to get a relative better buckets.

**Comment by kangkaisen** [comment]
@WingsGo   Yes, the strategy  should be Pluggable.  I think this issue is like HBase (Tikv) region auto merge and split, Kylin(Druid) segment merge. If we know the best size for one table(which could be configured), we could auto split  and merge Doris partitions by tmp partition mechanism.
